### PR TITLE
HPCC-28920 Roxie crash in DBGLOG from deblacklist()

### PR DESCRIPTION
**Pull request by richardkchapman**
Additional fix was needed.

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
Reproduced via additional tracing in thorsoapcall (left in but commented out) to make the problem easier to provoke, then running a query in server mode that made a soapcall to a non-existant port.

### CODE REVIEW

File: system/jlib/jthread.cpp [hunk 1]
```diff
@@ -874,6 +874,12 @@ class CPooledThreadWrapper: public Thread
     {
         parent.numrunning++;
     }
+    virtual void start() override
```
Comment (richardkchapman) [inline system/jlib/jthread.cpp:877]: Arguably a better fix is to move the capture of logctx from Thread::start into Thread::Thread constructor

**Review by mckellyln — APPROVED** [review]
Looks good.

**Comment by ghalliday** [comment]
Just to clarify - this fixes the problem because a thread may actually be started much later than it is created - in which case it will potentially refer to a context that then gets destroyed?

**Review by ghalliday — COMMENTED** [review]
Is a similar change required for CThreadedPersistent?

**Comment by ghalliday** [comment]
@richardkchapman

**Comment by mckellyln** [comment]
https://track.hpccsystems.com/browse/HPCC-28955

**Comment by richardkchapman** [comment]
In the particular instance that is crashing, the thread is started to asynchronously clean up (deblacklist a socket), so the thread that is active at the time is very likely to have terminated before the started thread is. The thread pool however is always going to outlive the threads in it.

There was code present to try to ensure that the context was captured at the time that the threadpool was created, but it didnt work properly. This change ensures that the context for threadpools IS captured properly at threadpool creation time, but also means the context for threads is captured at thread creation time rather than thread start time. THis is more likely to be safe, though neither is gruaranteed.

ThreadedPersistent are started at creation time so the change does not affect their behaviour.

**Review by mckellyln — APPROVED** [review]
Looks ok to merge

**Review by ghalliday — APPROVED** [review]

**Pull request merged**